src/main/java/io/debezium/connector/yugabytedb/transforms/StriimCompatible.java

```diff
@@ -0,0 +1,208 @@
+    }
+
+    @Override
+    public ConfigDef config() {
+        return new ConfigDef();
+    }
+
+    @Override
+    public void close() {
+    }
+
+    private List<String> getAllFieldsInOrder(Schema schema) {
+        List<String> fields = Arrays.asList(new String[schema.fields().size()]);
+        for (Field field : schema.fields()) {
+            fields.set(field.index(), field.name());
+        }
+        return fields;
+    }
+
+    private boolean isValueSetStruct(Field field) {
+        return field.schema().fields().size() == 2
+                && (Objects.equals(field.schema().fields().get(0).name(), "value")
+                && Objects.equals(field.schema().fields().get(1).name(), "set"));
+    }
+
+    private Schema makeMetadataSchema() {
+        final SchemaBuilder builder = SchemaBuilder.struct();
+        builder.field("LSN", Schema.STRING_SCHEMA);
+        builder.field("OperationName", Schema.STRING_SCHEMA);
+        builder.field("PK_UPDATE", Schema.OPTIONAL_BOOLEAN_SCHEMA);
+        builder.field("Sequence", Schema.STRING_SCHEMA);
+        builder.field("TableName", Schema.STRING_SCHEMA);
+        builder.field("TxnID", Schema.STRING_SCHEMA);
+        return builder.build();
+    }
+
+    private Struct makeMetadata(Struct value, Schema metadataSchema) {
+        Struct metadata = new Struct(metadataSchema);
+        Struct sourceValue = (Struct) value.get("source");
+        metadata.put("LSN", sourceValue.get("lsn").toString());
+        metadata.put("Sequence", sourceValue.get("sequence").toString());
+        metadata.put("TxnID", sourceValue.get("txId"));
+        metadata.put("TableName", sourceValue.get("schema") + "." + sourceValue.get("table"));
+        return metadata;
+    }
+
+    private Schema makeUpdatedSchema(Schema schema) {
+        final SchemaBuilder arrayBuilder = SchemaBuilder.array(Schema.OPTIONAL_STRING_SCHEMA);
+        final SchemaBuilder builder = SchemaUtil.copySchemaBasics(schema, SchemaBuilder.struct());
+        builder.field("metadata", makeMetadataSchema());
+        builder.field("data", arrayBuilder.build());
+        builder.field("columns", arrayBuilder.build());
+        builder.field("before", arrayBuilder.optional().build());
+        return builder.build();
+    }
+
+    private Map<String, Object> extractData(Struct value) {
+        Map<String, Object> values = new HashMap<>();
+        if (value == null) {
+            return values;
+        }
+
+        for (Field field : value.schema().fields()) {
+            LOGGER.debug("Considering value {}", field.name());
+            if (field.schema().type() == Type.STRUCT) {
+                LOGGER.debug("Value is a struct");
+                Struct fieldValue = (Struct) value.get(field);
+                if (isValueSetStruct(field)) {
+                    LOGGER.debug("value is valueset");
```

Comment:
The log doesn't make much sense if seen from the bigger picture, can you please improve it a little? Maybe mention that `Value while extracting data is a valueset` - or something similar.

Reply:
Removed these logs because there were logs already for printing the complete updated value. Actually, I added these logs originally when I was testing this transformer.